src/reducers/editSucessReducer.js

```diff
@@ -0,0 +1,19 @@
+import { fromJS } from 'immutable';
```

Comment:
Usually you would use a switch statement in reducers, because they generally get bigger when the application grows, and it's easier to understand them since there is not so much indentation. 

Regarding the name and types of actions this reducer is handling, I think you should move them to a more suitable place. When you create a reducer you have to decide which slice of the state that reducer is affecting. In this case it would be the user and the location for example. But I wouldn't create a reducer that is a only concerned with an action (edit) and one single outcome (success), besides by its name we don't know what had a successful edition, if it was a user, a location, or any other thing that could be edited. 
You can check out this part of the documentation from redux on how to structure reducers: https://redux.js.org/recipes/structuringreducers/basicreducerstructure